src/infoPanel/infoPanel.c

```diff
@@ -33,27 +38,32 @@ bool loadImagesPaths(const char* config_path, char ***images_paths, int *images_
 	cJSON* json_images_array = cJSON_GetObjectItem(json_root, "images");
 	*images_paths_count = cJSON_GetArraySize(json_images_array);
 	*images_paths = (char**)malloc(*images_paths_count * sizeof(char*));
+	*images_titles = (char**)malloc(*images_paths_count * sizeof(char*));
 
 	for (int i = 0; i < *images_paths_count; i++)
 	{
 		(*images_paths)[i] = (char*)malloc(STR_MAX * sizeof(char));
+		static const int g_title_max_length = 50;
+		(*images_titles)[i] = (char*)malloc(g_title_max_length * sizeof(char));
 
-		cJSON* json_image_path_item = cJSON_GetArrayItem(json_images_array, i);
-		cJSON* json_image_path = cJSON_GetObjectItem(json_image_path_item, "path");
+		cJSON* json_image_item = cJSON_GetArrayItem(json_images_array, i);
+		cJSON* json_image_path = cJSON_GetObjectItem(json_image_item, "path");
 		char* image_path = cJSON_GetStringValue(json_image_path);
-		strcpy((*images_paths)[i], image_path);
+		strncpy((*images_paths)[i], image_path, STR_MAX);
+
+		cJSON* json_image_title = cJSON_GetObjectItem(json_image_item, "title");
+		char* image_title = cJSON_GetStringValue(json_image_title);
+		strncpy((*images_titles)[i], image_title, g_title_max_length);
 	}
 
 	cJSON_free(json_root);
 
     return true;
 }
 
-void drawInfoPanel(SDL_Surface *screen, SDL_Surface *video, const char *title_str, char *message_str)
+static void drawInfoPanel(SDL_Surface *screen, SDL_Surface *video, const char *title_str, char *message_str)
 {
-	SDL_BlitSurface(theme_background(), NULL, screen, NULL);
-	SDL_BlitSurface(screen, NULL, video, NULL); 
-	SDL_Flip(video);
+	theme_backgroundBlit(video);
```

Comment:
You shouldn't be using this here. You need to blit the background to `screen` not `video`.